src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/ShellInProcess.cs

```diff
@@ -8,17 +8,65 @@
 using System.Reflection;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Windows.Controls;
+using System.Windows.Input;
 using Microsoft.CodeAnalysis.UnitTests;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Threading;
+using Xunit;
 using IAsyncDisposable = System.IAsyncDisposable;
 
 namespace Microsoft.VisualStudio.Extensibility.Testing
 {
     internal partial class ShellInProcess
     {
+        /// <returns>True if the AllInOneSearch is being used for Navigation</returns>
+        public async Task<bool> ShowNavigateToDialogAsync(CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            await TestServices.Shell.ExecuteCommandAsync(VSConstants.VSStd12CmdID.NavigateTo, cancellationToken);
+
+            return await WaitForNavigateToFocusAsync(cancellationToken);
+
+            async Task<bool> WaitForNavigateToFocusAsync(CancellationToken cancellationToken)
+            {
+                bool? isAllInOneSearchActive = null;
+
+                while (true)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    // Take no direct action regarding activation, but assert the correct item already has focus
+                    TestServices.JoinableTaskFactory.Run(async () =>
```

Comment:
Why calling .Run() here since we're already in an async method?

Reply:
I borrowed this code from the InputInProcess.SendToNavigateToAsync. Will admit that I need to learn more about JTF generally. I guess we could move this into the method and add a transition off the main thread following the control check.